oauth/.gitignore

```diff
@@ -0,0 +1,2 @@
+# @todo Is this the proper place for oauth keys?
```

Comment:
@modularlab No, I think this shouldn't be the part of the codebase. This should be out of the codebase ?
cc: @msherron / @fiasco

Reply:
The keys are ignored, and do need to live in the codebase (outside of docroot). but the oauth config does need to be pointed somewhere to reference the keys.

Reply:
@modularlab 

1. I think, since this would be out of drupal `docroot`, then this should be ok. As this path is not web-accessible. Right ? cc: @msherron Opinion ?

2. Also, I think we should create another sub-folder in this(with path or name of current site). Otherwise, if multi-site is setup, then this keys would get overwritten by them. As all multi-site would share the common folder. For this, I think we can use service `'site.path'` as this gives the path for the current site.

3. When I'm installing this module on vanilla drupal (i.e with either standard/minimal profile), then it's giving me following error: 
 ```
The database connection is not serializable. This probably means you are serializing an object that 
has an indirect reference to the database connection. Adjust your code so that is not necessary. 
Alternatively, look at DependencySerializationTrait as a temporary solution. in 
/Users/vishal.khode/Sites/headless/docroot/core/lib/Drupal/Core/Database/Connection.php on line 1993 #0 [internal function]: Drupal\Core\Database\Connection->__sleep()

``` 
When I debugged I found that I don't have `../oauth` directory (in my vanilla drupal codebase). So, I manually created this directory and error is gone.
  3.1 Can we write the code to check & create directory for oauth, in `hook_install()` so that I don't need to manually create directory ?
  3.2 Can we also log the error message in logger, so I can see the error/warnings from DB log ?(https://github.com/acquia/acquia_cms/pull/1116/files#diff-7cc0030c40f70980b5a527c1dea6f2e932c734be1317eb1489bb851a7041571fR197) 

4. When enabling headless mode, I'm seeing the following warning in my DB log. Can we fix this ?
```
Notice: Undefined index: parent in acquia_cms_headless_ui_menu_links_discovered_alter() (line 352 of /Applications/MAMP/htdocs/drupal/docroot/modules/contrib/acquia_cms_headless/modules/acquia_cms_headless_ui/acquia_cms_headless_ui.module)
```